syntax = "proto3";

import "google/protobuf/struct.proto";

package rpc_lib.config;

enum FlagType {
    UNDEFINED = 0;
    BOOLEAN = 1;
    INTEGER = 2;
    FLOAT = 3;
    STRING = 4;
    JSON = 5;
}

message FlagDefinition {
    string name = 1;
    FlagType type = 2;
    FlagValue value = 3;
}

message FlagValue {
    message Override {
        google.protobuf.Value value = 1;
        FlagCondition condition = 2;
    }

    google.protobuf.Value base_value = 1;
    repeated Override overrides = 2;
}


message FlagCondition {
    string condition_name = 1;
    reserved 2;
    map<string, google.protobuf.Value> params = 3;
}

